std/stdio.d

```diff
@@ -107,7 +107,11 @@ version (Windows)
 {
     private alias FSChar = wchar;
 }
-else version (Posix)
+else version (linux)
+{
+    private alias FSChar = char;
+}
+else version (GENERIC_IO)
```

Comment:
What's GENERIC_IO?

Reply:
It's set on a number of platforms. Look at the file. I don't really know how to review, because I don't know all those platforms.

Reply:
And if some platform isn't covered static assert will catch it

Reply:
Looking at the different platforms, the following set `GENERIC_IO`, and I don't know if there are any that aren't `version(Posix)`:

```
CRuntime_Bionic,
CRuntime_Musl,
CRuntime_UClibc,
OSX,
iOS,
TVOS,
WatchOS,
FreeBSD,
NetBSD,
DragonFlyBSD,
Solaris
```

So I'm not sure which platforms this is targeting. Or is this intended as a "just in case"? If so, I'd suspect that once that platform is added, it would need to be addressed when that platform is added.

Reply:
I use it for ARM semihosting output on non-posix bare metal device.

Reply:
> `"dwarfeh(364) fatal error"` is printed from druntime

and doesn't with this change?

Reply:
So what OS version is defined for that configuration?

Reply:
> and doesn't with this change?

yes

> Oh wait, it works, because of this change, and properly prints that. I got it.

It isn't compiles without this change, because here is not Windows and also not Posix version is set.

> So what OS version is defined for that configuration? And what version is defined that adds GENERIC_IO. In other words, what is the version configuration that applies to this case.

I use ldc2 with `--mtriple=thumbv7m-unknown-none-eabi` option. Looks like no OS version provided.

But here is available druntime and not posix-compilant libc.

(Also it is impossible to pass Posix version into compiler - it forbids passing of built-in versions values.)

Reply:
when you run ldc2 -v what does it list for `predefs`?

Reply:
DruntimeAbstractRt CRuntime_Abstract AsmExternal GENERIC_IO CoreUnittest LDC all D_Version2 unittest assert D_NoBoundsChecks D_ModuleInfo D_Exceptions D_TypeInfo ARM Thumb ARM_Thumb ARM_SoftFP D_HardFloat LittleEndian LDC_LLVM_1000

Reply:
Wow, so `GENERIC_IO` is added by ldc2? that surprises me. And I think I have to agree with this change then.

Reply:
> Wow, so GENERIC_IO is added by ldc2?

No, I added it from cmd line while compiling

Reply:
OK, that makes sense. Yeah, I understand now. Honestly, with no defined OS, it should make sense to be able to configure whichever way you want. I'm OK with the changes then.